readme_docs/VOYAGE_STYLE_GUIDE.md

```diff
@@ -0,0 +1,2237 @@
+createHero(
+  firstName,
+  lastName,
+  inventorOf
+);
+
+// good
+createHero(
+  firstName,
+  lastName,
+  inventorOf,
+);
+
+// good (note that a comma must not appear after a "rest" element)
+createHero(
+  firstName,
+  lastName,
+  inventorOf,
+  ...heroArgs
+)
+```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Semicolons
+
+**Yup.**
+
+```typescript
+// bad
+(function () {
+  const name = 'Skywalker'
+  return name
+})()
+
+// good
+(function () {
+  const name = 'Skywalker';
+  return name;
+}());
+
+// good, but legacy (guards against the function becoming an argument when two files with IIFEs are concatenated)
+;(() => {
+  const name = 'Skywalker';
+  return name;
+}());
+```
+
+    [Read more](https://stackoverflow.com/questions/7365172/semicolon-before-self-invoking-function/7365214%237365214).
+
+**[⬆ back to top](#table-of-contents)**
+
+## Type Casting & Coercion
+
+#### Perform type coercion at the beginning of the statement
+
+- ##### Strings:
+
+    ```typescript
+    // => this.reviewScore = 9;
+
+    // bad
+    const totalScore = this.reviewScore + ''; // invokes this.reviewScore.valueOf()
+
+    // bad
+    const totalScore = this.reviewScore.toString(); // isn't guaranteed to return a string
+
+    // good
+    const totalScore = String(this.reviewScore);
+    ```
+
+- ##### Numbers: Use `Number` for type casting and `parseInt` always with a radix
+
+    ```typescript
+    const inputValue = '4';
+
+    // bad
+    const val = new Number(inputValue);
+
+    // bad
+    const val = +inputValue;
+
+    // bad
+    const val = inputValue >> 0;
+
+    // bad
+    const val = parseInt(inputValue);
+
+    // good
+    const val = Number(inputValue);
+
+    // good
+    const val = parseInt(inputValue, 10);
+    ```
+
+
+- ##### Booleans:
+
+    ```typescript
+    const age = 0;
+
+    // bad
+    const hasAge = new Boolean(age);
+
+    // good
+    const hasAge = Boolean(age);
+
+    // best
+    const hasAge = !!age;
+    ```
+
+**[⬆ back to top](#table-of-contents)**
+
+
+## Naming Conventions
+
+#### **NEVER** single letter names. **BE DESCRIPTIVE**
+
+```typescript
+// bad
+function q() {
+  // ...stuff...
+}
+
+// good
+function queryDatabase() {
+  // ..stuff..
+}
+```
+
+#### Use camelCase when naming objects, functions, and instances
+
+```typescript
+// bad
+const OBJEcttsssss = {};
+const this_is_my_object = {};
+function c() {}
+
+// good
+const thisIsMyObject = {};
+function thisIsMyFunction() {}
+```
+
+#### Do not use trailing or leading underscores
+
+  > Why? JavaScript does not have the concept of privacy in terms of properties or methods. Although a leading underscore is a common convention to mean “private”, in fact, these properties are fully public, and as such, are part of your public API contract. This convention might lead developers to wrongly think that a change won't count as breaking, or that tests aren't needed. tl;dr: if you want something to be “private”, it must not be observably present.
+
```

Comment:
This section is rendering the triple backticks. Maybe because of the indenting of the section?